### PR TITLE
DuneSQL Migration: new macros, new dunesql CI, and translated + legacy evms_blocks model

### PR DESCRIPTION
**Pull request by couralex6**
Trino macro prior to trino migration

### CODE REVIEW

File: models/uniswap/ethereum/uniswap_ethereum_sources.yml [hunk 1]
```diff
@@ -2,6 +2,7 @@ version: 2
 
 sources: 
   - name: uniswap_ethereum
+    database: delta_prod
```
Comment (jkylling) [inline models/uniswap/ethereum/uniswap_ethereum_sources.yml:5]: remove

File: models/uniswap/ethereum/uniswap_v1_ethereum_trades.sql [hunk 2]
```diff
@@ -22,12 +23,12 @@ WITH dexs AS
     SELECT
         t.evt_block_time AS block_time
         ,t.buyer AS taker
-        ,'' AS maker
+        ,0x as maker
```
Comment (jkylling) [inline models/uniswap/ethereum/uniswap_v1_ethereum_trades.sql:26]: null

File: models/uniswap/ethereum/uniswap_v1_ethereum_trades.sql [hunk 3]
```diff
@@ -46,11 +47,11 @@ WITH dexs AS
...
```
Comment (jkylling) [inline models/uniswap/ethereum/uniswap_v1_ethereum_trades.sql:50]: null

File: .github/workflows/dbt_slim_ci.yml [hunk 4]
```diff
@@ -37,28 +55,28 @@ jobs:
...
 
       - name: dbt run incremental model(s) if applicable
         if: env.INC_MODEL_COUNT > 0
-        run: "dbt run --select state:modified,config.materialized:incremental --exclude tag:prod_exclude --defer --state ."
+        run: "dbt run $PROFILE --select state:modified,config.materialized:incremental,$TAG --exclude $EXCLUDEtag:prod_exclude --defer --state ."
 
       - name: dbt test incremental model(s) if applicable
         if: env.INC_MODEL_COUNT > 0
-        run: "dbt test --select state:modified,config.materialized:incremental --exclude tag:prod_exclude --defer --state ."
+        run: "dbt test $PROFILE --select state:modified,config.materialized:incremental,$TAG --exclude $EXCLUDEtag:prod_exclude --defer --state ."
```
Comment (jkylling) [inline .github/workflows/dbt_slim_ci.yml:79]:
```suggestion
        run: "dbt test $PROFILE --select state:modified,config.materialized:incremental,$TAG --exclude ${EXCLUDE}tag:prod_exclude --defer --state ."
```

File: macros/expose_spells.sql [hunk 5]
```diff
@@ -1,15 +1,29 @@
...
-        'dune.public'='true',
-        'dune.data_explorer.blockchains'= '{{ blockchains }}',     -- e.g., ["ethereum","solana"]
-        'dune.data_explorer.category'='abstraction',
-        'dune.data_explorer.abstraction.type'= '{{ spell_type }}', -- 'project' or 'sector'
-        'dune.data_explorer.abstraction.name'= '{{ spell_name }}', -- 'aave' or 'uniswap'
-        'dune.data_explorer.contributors'= '{{ contributors }}',   -- e.g., ["soispoke","jeff_dude"]
-        'dune.vacuum' = '{"enabled":true}'
-        )
-{%- else -%}
+{%- if target.name == 'prod' or True-%}
```
Comment (jkylling) [inline macros/expose_spells.sql:2]: Remove `or True`

File: macros/mark_as_spell.sql [hunk 6]
```diff
@@ -1,9 +1,17 @@
 {% macro mark_as_spell(this, materialization) %}
-{%- if target.name == 'prod'-%}
-        ALTER {{"view" if materialization == "view" else "table"}} {{ this }}
-        SET TBLPROPERTIES (
-        'dune.data_explorer.category'='abstraction'
-        )
-{%- else -%}
+{%- if target.name == 'prod' or True-%}
```
Comment (jkylling) [inline macros/mark_as_spell.sql:2]: Remove `or True`

File: .github/workflows/commit_manifest.yml [hunk 7]
```diff
@@ -21,25 +27,40 @@ jobs:
...
+          echo "Unknown engine: ${{ matrix.engine }}"
+          exit 1
+        fi
+    
     - name: dbt dependencies
       run: "dbt deps"
 
     - name: dbt compile to create prod manifest from main
-      run: "dbt compile --target-path ."
+      run: "dbt compile --target-path . $PROFILE $COMPILE_TAG"
```
Comment (jkylling) [inline .github/workflows/commit_manifest.yml:53]: This needs an exclude

File: macros/dune/schema.sql [hunk 8]
```diff
@@ -0,0 +1,10 @@
+{% macro create_schema(relation) -%}
+  {{ adapter.dispatch('create_schema', 'dbt')(relation) }}
+{% endmacro %}
+
+{% macro default__create_schema(relation) -%}
+  {% set s3_bucket = 'trino-dev-datasets-118330671040' %}
```
Comment (jkylling) [inline macros/dune/schema.sql:6]: Is this bucket correct?

**Review by jkylling — APPROVED** [review]

**Comment by jeff-dude** [comment]
taking a few notes here to prep for merge, mostly for myself to follow:
- modifying existing macros, especially `mark_as_spell` but also `expose_spells` to a lesser extent, typically make dbt think an entire full refresh is needed, as they impact all spells -- **_therefore, i'd recommend we disable the `state:modified` step in our orchestration for this merge, so we avoid unnecessary full refresh on all 1k+ spells_**
- we should monitor PRs closely after merge to ensure `commit_manifest` workflow runs as expected with engine profiles and env variables added -- **_no real takeaway here other than be cognizant of PR activity after to ensure we see results as expected, as it'll impact all PR CI tests due to manifest comparisons and running only needed models per PR_**
- ensure `DBT_ENV_CUSTOM_ENV_S3_BUCKET` is set as expected in dbt cloud for new trino project
- ensure we educate users on how to leverage new `alias` config property across engines
- is [this](https://github.com/duneanalytics/spellbook/pull/3570/files#diff-e245c6d08a50f784f8b0c87debaef35c60a10f16a1651509353c123a7e65b374R6) our bucket for both dev PRs and prod?

**Comment by jeff-dude** [comment]
not critical path, but maybe nice-to-have:
- for the matrix engine, have both jobs (spark and dunesql) run simultaneously rather than sequential?

**Comment by dot2dotseurat** [comment]
regarding permissions bug, this is the [PR](https://github.com/duneanalytics/infra/pull/1120) from the s3 permissions I initially set up on the original manifest bucket. have we made a similar pr for the new bucket `manifest-spellbook-dunesql`?

**Comment by hildobby** [comment]
Hey @couralex6, I'm looking into adding Celo to the tables in the `evms` sector. What are the reasons for the creation of `evms_blocks_legacy.sql` as a separate file from `evms_blocks.sql`? Seems like the added `'dunesql'` tag and the altered alias are the only two differences. What is the purpose of this change and why this specific model? Which file should I build on top of now?


Screenshot 2023-06-28 at 16 27 54
re the differences

**Comment by hildobby** [comment]
> > Hey @couralex6, I'm looking into adding Celo to the tables in the `evms` sector. What are the reasons for the creation of `evms_blocks_legacy.sql` as a separate file from `evms_blocks.sql`? Seems like the added `'dunesql'` tag and the altered alias are the only two differences. What is the purpose of this change and why this specific model? Which file should I build on top of now?
> > Screenshot 2023-06-28 at 16 27 54
> 
> we are still working on an updated contribution guide that will include most of these details for yourself and others. if you're looking to get ahead, here are some answers:
> 
> * `_legacy.sql` files will maintain the spark logic and continue to run on spark engine, so the data maintains in sync until we deprecate spark completely. also within legacy file will be updated alias property in config, to match new alias macro
> * the translated to dunesql files will not contain `_legacy` and add a few things to help orchestration and table naming. anything written in dunesql syntax will need that tag for all orchestration needs. the alias macro (i.e. what the property you're referring to calls) has been edited to accept more params, including legacy flag, to help name tables with or without the legacy suffix.
> * this specific model was chosen since it's a view and quick to run in CI tests and was top of mind for me :)
> * build on both files for the time being, hopefully that will be short lived process to edit both
> 
> we will ensure to look closely in review on new PRs, then can use findings to help improve contribution guide

Ah alright makes sense! I made a PR to add Celo but looks like it passed tests without needing me to touch anything new, lmk if there's something new that ci tests dont account for! https://github.com/duneanalytics/spellbook/pull/3640

**Comment by jeff-dude** [comment]
> a PR to add Celo but looks like it passed tests without needing me to touc

will follow up on that PR when i can 🙏

**Review by alokeutpal — APPROVED** [review]

**Pull request merged**